### DISCUSSION TITLE
Start with windows issue and can ports be split tunneled?

### DISCUSSION THREAD

**Discussion by scrollop** [wiresock/WireSockUI#40] [General]
Hello,
I love this app - it's so useful!

1) Prior to this release, when I added winsockUI to the startup list, it didn't start. Couldn't figure out why, so I was happy to see a "Start with windows" option in settings, however on my win11 box, after I tick this box and save and exit settings, when I re-0enter settings the box is unticked.

2) I find that after a while (various times), the VPN stalls (not sure if all traffic stops, though I can't access the computer on the LAN via remote desktop or samba).

3) Is there a way to bypass a port (eg samba and RDP ports) as we can to bypass programs and addresses?

Thank you!

**Comment by wiresock** [comment]
Hello,

Thank you for your valuable feedback!

1. The "Start with Windows" feature in WireSockUI, though functional, is a workaround involving the Task Scheduler to circumvent UAC limitations. It's not the most ideal method and may lead to conflicts if WireSockUI is launched from different locations. To ensure smooth operation, I advise closing WireSockUI, opening Task Scheduler, and if a WireSockUI task exists, delete it. Then, relaunch WireSockUI and activate the "Start with Windows" feature.

2. The issue you're encountering could stem from various factors. To diagnose it effectively, I need more details, such as the operating mode (with or without a virtual network interface), the type of connection (WiFi or wired), your configuration settings, and, ideally, a log from the CLI version of WireSock VPN Client captured at the time of the issue.

3. Currently, WireSock VPN Client does not support port-based split tunneling. Although it's a feasible addition, it might overly complicate the system's logic. I'll consider implementing it after further evaluation.

Cheers,
Vadim

**Reply by e-t-l** [reply]
Hey Vadim, so does the "start with windows" option mean that WireSock will connect to a VPN before any user has logged in? (I am a device admin if that makes any difference. My goal is for my VPN to start as soon as the device starts, before I log on to my account)

**Reply by e-t-l** [reply]
...actually, unless WireSockUI does something different at a system level, I think https://github.com/TunnlTo/desktop-app/discussions/34 is probably the discussion I need to be watching if I want WireSock to launch at startup, right?

**Reply by wiresock** [reply]
The "start with windows" option typically means that the application will launch automatically when a user logs into their Windows account. However, this does not imply that the application starts running and establishes a VPN connection before any user logs in.

In your case, since you need the VPN connection to be active right from the system startup (before any user logs in), running WireSock as a standard application with the "start with windows" feature won't meet your requirement.

For your specific need, I recommend using WireSock VPN Client as a Windows Service. Running VPN software as a Windows Service allows the VPN connection to be established at the system startup level, before any user logs into the device. This ensures that your VPN is active and your network traffic is secured right from the moment your device starts.

`wiresock-client.exe install -start-type 2 -config [config_full_path_name] -log-level none`

**Reply by e-t-l** [reply]
Thank you. And WireSock-client still permits split tunneling, right?

**Reply by wiresock** [reply]
Yes, sure.

**Comment by scrollop** [comment]
Thanks for your help.

1) That worked! The start with windows option stays selected when re entering settings. Will see if it loads with windows.

2) I will log and report back.

Thanks again!
